frontend/app/src/components/DateInput.svelte

```diff
@@ -20,57 +28,83 @@
 
     import { _ } from "svelte-i18n";
     import { interpolate } from "../i18n/i18n";
-    import { DateInput } from "date-picker-svelte";
+    import SveltyPicker from "svelty-picker";
     import type { ResourceKey } from "openchat-client";
-    import { onMount } from "svelte";
+    import { onMount, onDestroy } from "svelte";
 
     let {
         align = "left",
-        closeOnSelection = false,
+        clearBtn = true,
+        clearToggle = true,
         disabled = false,
-        format = "dd MMM yyyy, HH:mm",
+        endDate = undefined,
+        format = "dd M yyyy, hh:ii",
         futureOnly = false,
+        inputId = undefined,
+        inputClasses = "",
+        minuteIncrement = 1,
+        mode = "datetime",
+        pickerOnly = false,
         placeholder = undefined,
         required = false,
-        timePrecision = "minute",
-        valid = true,
+        startDate = undefined,
+        todayBtn = true,
         value = undefined,
-        onselect,
+        weekStart = 1,
+        onchange = undefined,
     }: Props = $props();
 
     // If we're only allowed to select future dates, then minimum date and
     // time selectable is current!
-    let minDate: Date | undefined = futureOnly ? new Date() : undefined;
-
-    // Allow selection of up to 10 years in the future.
-    let maxDate = new Date();
-    maxDate.setFullYear(maxDate.getFullYear() + 10);
+    startDate = futureOnly ? new Date() : undefined;
```

Comment:
I don't think we should have this complexity. I think we should just allow any date/time to be picked but then show a validation error next to the text input if it is in the future